airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -829,10 +829,12 @@ def _list(
                     versions=versions,
                 )
 
+            all_blobs = list(blobs)
```

Comment:
Since this issue is specific to using `prefix` and `delimiter`, it would be better for future debugging to convert it into `blobs = list(blobs)` within the `else` block above (and revert the later reference).

Reply:
@shahar1 thanks for taking a look at this, much appreciated. I've moved the `list(blobs)` into the above `else` as requested. I tested this locally using `prefix` with `match_glob` and then, `prefix` with `delimiter` and both code paths appear to work fine.

Reply:
I noticed this was also broken for `list_by_timespan`, I've fixed that as well and squashed the commits.